dev/archery/archery/bot.py

```diff
@@ -82,6 +87,121 @@ def parse_args(self, ctx, args):
+        return self.github.get_repo(self.event_payload['repository']['id'], lazy=True)
+
+    def handle(self):
+        current_state = None
+        try:
+            current_state = self.get_current_state()
+        except EventError:
+            # In case of error (more than one state) we clear state labels
+            # only possible if a label has been manually added.
+            self.clear_current_state()
+        new_state = self.get_target_state(current_state)
+        if current_state != new_state.value:
+            if current_state:
+                self.clear_current_state()
+            self.set_state(new_state)
+
+    def get_current_state(self):
+        """
+        Returns a string with the current PR state label
+        based on label starting with LABEL_PREFIX.
+        If more than one label is found raises EventError.
+        If no label is found returns None.
+        """
+        states = [label.name for label in self.pull.get_labels()
+                  if label.name.startswith(LABEL_PREFIX)]
+        if len(states) > 1:
+            raise EventError(f"PR cannot be on more than one states - {states}")
+        elif states:
+            return states[0]
+
+    def clear_current_state(self):
+        """
+        Removes all existing labels starting with LABEL_PREFIX
+        """
+        for label in self.pull.get_labels():
+            if label.name.startswith(LABEL_PREFIX):
+                self.pull.remove_from_labels(label)
+
+    def get_target_state(self, current_state):
+        """
+        Returns the expected target state based on the event and
+        the current state.
+        """
+        if (self.event_name == "pull_request_target" and
+                self.event_payload['action'] == 'opened'):
+            if (self.event_payload['pull_request']['author_association'] in
+                    COMMITTER_ROLES):
+                return PullRequestState.committer_review
+            else:
+                return PullRequestState.review
+        elif (self.event_name == "pull_request_review" and
+                self.event_payload["action"] == "submitted"):
+            review_state = self.event_payload["review"]["state"].lower()
+            is_committer_review = (self.event_payload['review']['author_association']
+                                   in COMMITTER_ROLES)
+            if not is_committer_review:
+                # Non-committer reviews cannot change state once committer has already
+                # reviewed and requested changes.
+                if current_state in (
+                        PullRequestState.change_review.value,
+                        PullRequestState.changes.value):
+                    return PullRequestState(current_state)
+                else:
+                    return PullRequestState.committer_review
+            if review_state == 'approved':
+                return PullRequestState.merge
+            elif review_state in ("changes_requested", "commented"):
```

Comment:
Can we simplify this?

```suggestion
            else
```